src/tests/ftest/rebuild/container_create_race.py

```diff
@@ -1,284 +1,166 @@
+        dkey_size = self.params.get("dkey_size", "/run/io/*")
+        cont_oclass = self.params.get("oclass", "/run/io/*")
         count = 0
-        while not pool1.rebuild_complete() and count < qty:
-            # Create a new container
+        self.container = []
+        while not self.pool.rebuild_complete() and count < qty:
             count += 1
             self.log.info(
-                "%s: Creating container %s/%s in pool %s during rebuild",
-                loop_id, count, qty, pool2.uuid)
-            self.container.append(TestContainer(pool2))
-            self.container[-1].get_params(self)
-            self.container[-1].create()
-            self.container[-1].write_objects()
-
-        if count < qty:
-            self.fail(
-                "{}: Rebuild completed with only {}/{} containers "
-                "created".format(loop_id, count, qty))
-
-    def run_ior(self, loop_id, mpirun):
-        """Run the ior command defined by the specified ior command object.
-
-        Args:
-            loop_id (str): loop identification string
-            mpirun (Mpirun): mpirun command object to run ior
-        """
-        total_bytes = mpirun.job.get_aggregate_total(mpirun.processes.value)
-        try:
-            mpirun.run()
-        except CommandFailure as error:
-            self.fail(
-                "{}: Error populating the container with {} bytes of data "
-                "prior to target exclusion: {}".format(
-                    loop_id, total_bytes, error))
-        self.log.info(
-            "%s: %s %s bytes to the container",
-            loop_id, "Wrote" if "-w" in mpirun.job.flags.value else "Read",
-            total_bytes)
-
-    def access_container(self, loop_id, index, message):
+                "..Creating container %s/%s in pool %s during rebuild",
+                count, qty, self.pool.uuid)
+            self.container.append(self.get_container(self.pool))
+
+        self.log.info("..Create %s containers and write data during rebuild.",
+                      qty)
+        self.container[index].object_qty.value = object_qty
+        self.container[index].record_qty.value = record_qty
+        self.container[index].data_size.value = data_size
+        self.container[index].akey_size.value = akey_size
+        self.container[index].dkey_size.value = dkey_size
+        self.container[index].oclass.value = cont_oclass
+        self.container[index].write_objects(rank, cont_oclass)
+
+    def access_container(self, index=-1):
         """Open and close the specified container.
-
         Args:
-            loop_id (str): loop identification string
             index (int): index of the daos container object to open/close
-            message (str): additional text describing the container
-
         Returns:
             bool: was the opening and closing of the container successful
 
         """
         status = True
+        container = self.container[index]
```

Comment:
Same as above.
Any reason you used index 1? We could have used default index=0 for iterating the container array... Is it not?

Reply:
same as above